### ISSUE TITLE
Deprecate ContextMark.__call__() and force the use of explicit ContextMark.attach() and ContextMark.detach() instead

### ISSUE THREAD

**Issue by trevorbaca** [Abjad/abjad#264]
_From [trevorb...@gmail.com](https://code.google.com/u/118334091018461915321/) on October 13, 2013 13:58:25_

The 2.14 way of doing things will move from ...

   clef.attach(staff)

... to ...

   attach(clef, staff)

... instead.

So forcing the use of clef.attach(staff) instead of clef(staff) will help us with the eventual migration.

Also, the idiom of clef(staff) is always confusing to newcomers.

_Original issue: http://code.google.com/p/abjad/issues/detail?id=264_
